src/Compilers/Core/Portable/Compilation/DeterministicKeyBuilder.cs

```diff
@@ -185,8 +185,11 @@ void writeGenerators()
                 writer.WriteArrayStart();
                 foreach (var generator in generators)
                 {
+                    object obj = generator is IncrementalGeneratorWrapper w
```

Comment:
We have an extension method for this exact scenario: https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/SourceGeneration/GeneratorExtensions.cs#L23